lib/ExpensiMark.js

```diff
@@ -468,6 +468,24 @@ export default class ExpensiMark {
                     match[2] = match[2].substr(0, match[2].length + brace);
                 }
             }
+
+            // Because we are removing ) parenthese, some special characters that shouldn't be in the href are in the href
+            // For example google.com/toto.) is accepted by the regexp, above we remove the ) parenthese so the link becomes google.com/toto. which is not a valid link
+            // In that case we should also remove the "."
+            var numberOfCharsToRemove = 0;
+
+            for (let i = match[2].length - 1; i >= 0; i--) {
+                if ("$*.+!(,=".includes(match[2][i])) {
+                    numberOfCharsToRemove++;
+                }
+                else
+                    break;
+            }
+            if (numberOfCharsToRemove){
+                match[0] = match[0].substr(0, match[0].length - numberOfCharsToRemove);
+                match[2] = match[2].substr(0, match[2].length - numberOfCharsToRemove);
+            }
+
```

Comment:
There is probably a way to make that work in the same code but I don't think it makes sense to try to do it because the code for `)` has to count the parentheses in the left to right order while this code starts from the end.
Also this situation where we have the special chars at the end only happens after we have remove the `)`

I am adding the special characters in CONST right now